.tekton/create-ci-image-pipelinerun.yaml

```diff
@@ -0,0 +1,118 @@
+          results:
+            - name: image-name
+            - name: image-name-timestamp
+          steps: 
+            - name: generate-image-name
+              image: quay.io/openshift-pipeline/ci
+              script: | 
+                #!/usr/bin/env bash
+                BRANCH_NAME={{ target_branch }}
+                IMAGE_NAME=""
+                IMAGE_NAME_TIMESTAMP=""
+                TIMESTAMP=$(date +"%y%m%d%H%M")
+
+                if [ "$BRANCH_NAME" == "master" ]; then 
+                  if [ {{ event_type }} == "pull_request" ]; then
+                      IMAGE_NAME="image-registry.openshift-image-registry.svc:5000/pipelines-ci/ci:latest"
+                      IMAGE_NAME_TIMESTAMP="image-registry.openshift-image-registry.svc:5000/pipelines-ci/ci:5.0-$TIMESTAMP"
+                  else
+                      IMAGE_NAME="quay.io/openshift-pipeline/ci:latest"
+                      IMAGE_NAME_TIMESTAMP="quay.io/openshift-pipeline/ci:5.0-$TIMESTAMP"
+                  fi
+                elif [[ "$BRANCH_NAME" == release-* ]]; then 
+                  if [ {{ event_type }} == "pull_request" ]; then
+                    IMAGE_NAME="image-registry.openshift-image-registry.svc:5000/pipelines-ci/ci:${BRANCH_NAME#release-}"
+                  else
+                    IMAGE_NAME="quay.io/openshift-pipeline/ci:${BRANCH_NAME#release-}"
+                  fi
+                  IMAGE_NAME_TIMESTAMP="$IMAGE_NAME-$TIMESTAMP"
+                else
+                    echo "Error: Branch name '$BRANCH_NAME' is not appropriate."
+                    exit 1
+                fi
+
+                echo "$IMAGE_NAME" | tee $(results.image-name.path)
+                echo "$IMAGE_NAME_TIMESTAMP" | tee $(results.image-name-timestamp.path)
+        runAfter: 
+          - fetch-repository
+      - name: buildah 
+        taskRef:
+          kind: ClusterTask
+          name: buildah
+        params:
+          - name: IMAGE
+            value: $(tasks.generate-image-name.results.image-name)
+          - name: DOCKERFILE
+            value: ./Dockerfile.CI
+        runAfter: 
+          - generate-image-name
+        workspaces:
+          - name: source
+            workspace: source
+      - name: skopeo-copy
+        taskRef: 
+          kind: ClusterTask
+          name: skopeo-copy
+        params:
+          - name: srcImageUrl
+            value: $(tasks.generate-image-name.results.image-name)
+          - name: destImageUrl
+            value: $(tasks.generate-image-name.results.image-name-timestamp)
+        runAfter:
+          - buildah
+    finally:
+      - name: send-slack-notification
+        taskRef:
+          name: send-slack-notification
+        params:
+          - name: MESSAGE
+            value: "<icon> Uploading CI image *<run_status>* <icon> <<logs_url>|logs>"
+        when:
+          - cel: "'$(tasks.buildah.status)' in ['Failed', 'None']"
+          - cel: "'$(tasks.skopeo-copy.status)' in ['Failed', 'None']"
```

Comment:
There is logical `and` betwen two `when` expressions so it wouldn't work as expected.

@VeereshAradhya If I remember correctly, there is some bug in Tekton that two CEL expression will not work, right? Do we need to combine both of those into one CEL?